### PR TITLE
Added multi-axis advanced indexing support

### PR DESCRIPTION
**Pull request by mikeymezher**
Added support for multi-axis advanced indexing. Speed is comparable to single axis advanced indexing.

### CODE REVIEW

**Comment by hameerabbasi** [comment]
Thanks! This is missing some tests, those are pretty much essential.

**Comment by mikeymezher** [comment]
It seems like the only failed test is test_slicing_errors(index). 
Seems like an issue with the tests. This is likely because an IndexError is no longer being raised if len(adv_idx) is greater than 1 in _coo.indexing.py

What's the protocol for modifying the test cases? 

Would you be able to handle this?

**Comment by hameerabbasi** [comment]
> Seems like an issue with the tests. This is likely because an IndexError is no longer being raised if len(adv_idx) is greater than 1 in _coo.indexing.py

That's because the test was "testing" if we didn't have advanced indexing at all. You can get rid of the bad tests.

**Comment by hameerabbasi** [comment]
> What's the protocol for modifying the test cases?

The contributor usually adds tests, and the maintainer/reviewer reviews the code and tests.

**Comment by mikeymezher** [comment]
Alright, I'll add a few test cases to test_slicing(index) and remove test_slicing_error are submit a new PR

**Comment by hameerabbasi** [comment]
Oh, you can just push to your branch and it'll modify this PR. Don't need to close at all.

**Comment by mikeymezher** [comment]
Bug found in returned coordinates. Will attempt to fix tomorrow

File: multi_adv_ix_test.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import sparse
```
Comment (hameerabbasi) [inline multi_adv_ix_test.py:1]: Can we remove this file?

File: multi_adv_ix_test2.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import sparse
```
Comment (hameerabbasi) [inline multi_adv_ix_test2.py:1]: And this one?

File: multi_adv_ix_test.py (hunk 1 above)
Comment (mikeymezher) [reply multi_adv_ix_test.py]: Yes sorry, I removed them in latest commit. Those were there just for local testing and I had forgot to remove them.

Comment (hameerabbasi) [reply multi_adv_ix_test.py]: No need to apologize. 😄 I tend to be a bit too quick in mt reviews.

File: sparse/_coo/indexing.py [hunk 3]
```diff
@@ -387,6 +464,8 @@ def _compute_mask(coords, indices):  # pragma: no cover
     n_matches = np.intp(coords.shape[1])
 
     i = 0
+    #print('COORD SHAPE: {}'.format(coords.shape))
+    #print('IXS SHAPE: {}'.format(indices.shape))
```
Comment (hameerabbasi) [inline sparse/_coo/indexing.py:468]: Can we remove these lines?

**Comment by hameerabbasi** [comment]
Thanks, @mikeymezher!

**Pull request merged**